File: java/ticket_office/src/main/java/fr/arolla/trainreservation/ticket_office/domain/Booking.java
package fr.arolla.trainreservation.ticket_office.domain;

import java.util.List;

/**
 * Represents a BookingRequest being processed.
 * It has the same fields as the BookingRequest, plus a reference
 * that serves as an Id
 */
public record Booking(String reference, String trainId, List<SeatId> seatIds) {
}

File: ts/ticket-office/domain/Booking.ts
import { SeatId } from "./Seat"

export default class Booking {
  bookingReference: string
  seatIds: SeatId[]
  trainId: string
  constructor(bookingReference: string, seatIds: SeatId[], trainId: string) {
    this.bookingReference = bookingReference
    this.seatIds = seatIds
    this.trainId = trainId
  }
}